### ISSUE TITLE
Running a generator throws error in terminal before the app even has a chance to send yo:answer event.

### ISSUE THREAD

**Issue by ayesham** [Signiant/socket-adapter#1]
I'm not sure if I am running the generator correctly so here is the code I have setup.

```javascript
env.lookup(function () {
   env.run('mygenerator', function (err) {
      console.log('Error:', err);
    });
});
```

Error:

```javascript
events.js:160
      throw er; // Unhandled 'error' event
      ^

TypeError: Cannot read property 'then' of undefined
    at child.Base.prompt (/Users/ayesha/Development/mygenerator/node_modules/yeoman-generator/lib/base.js:232:44)
    at child.prompting (/Users/ayesha/Development/mygenerator/generators/subgenerator/index.js:32:15)
    at Object.<anonymous> (/Users/ayesha/Development/mygenerator/node_modules/yeoman-generator/lib/base.js:431:23)
    at /Users/ayesha/Development/mygenerator/node_modules/run-async/index.js:25:25
    at /Users/ayesha/Development/mygenerator/node_modules/run-async/index.js:24:19
    at /Users/ayesha/Development/mygenerator/node_modules/yeoman-generator/lib/base.js:432:9
    at runCallback (timers.js:649:20)
    at tryOnImmediate (timers.js:622:5)
    at processImmediate [as _immediateCallback] (timers.js:594:5)
```

**Comment by jseed** [comment]
Looks like yeoman has evolved!  When I wrote this code yeoman generators used callbacks, not promises.  It looks to my like the error is a result of the .prompt function not returning a promise.  I will look into it further, should be a quick fix as long as it doesn't break any of our existing implementations.  
Thanks for the issue!

**Comment by jseed** [comment]
This should be now be fixed, make sure to update to version 0.3.0

**Comment by ayesham** [comment]
Works perfectly now! Thanks for the quick fix. 👍🏽
